### ISSUE TITLE
Flutter's CDN seems to be facing issues; I couldn't switch channels or run --doctor or do a fresh install.

### ISSUE THREAD

**Issue by Jayasubin** [flutter/flutter#146758]
### Steps to reproduce

1. Run the `flutter channel <channel name>` command.
   (or the `flutter --doctor` command)
   (or try downloading the flutter zip from the website)

### Actual results

Any flutter command will start downloading artefacts from flutter's CDNs.
The download speed will be extremely slow (around 10 kbps).
It times out after waiting for a long time.

Similar thing with zip download.

Is there any way i can download flutter other than the default cdn?

### Logs

<details open>
<summary>Logs</summary>

```console

~ % flutter --doctor

Downloading Darwin x64 Dart SDK from Flutter engine e76c956498841e1ab458577d3892003e553e4f3c...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0  214M    0  408k    0     0  37658      0  1:39:19  0:00:11  1:39:08 33096 2  2  214M    2 6529k    0     0  72620      0  0:51:30  0:01:32  0:49:58 97978

curl: (56) Recv failure: Operation timed out

/Users/<user>/dev/flutter/bin/internal/update_dart_sdk.sh: line 157: return: can only `return' from a function or sourced script
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:18 --:--:--     0
curl: (56) Recv failure: Operation timed out

Failed to retrieve the Dart SDK from: https://storage.googleapis.com/flutter_infra_release/flutter/e76c956498841e1ab458577d3892003e553e4f3c/dart-sdk-darwin-x64.zip

If you're located in China, please see this page:
  https://flutter.dev/community/china    

```

</details>

### Flutter Doctor output

<details open>
<summary>Doctor output</summary>

```console

wasn't able to get flutter doctor output as the prerequisite command doesn't run

```

</details>

**Comment by darshankawar** [comment]
> ```
> return: can only `return' from a function or sourced script
> ```

This is probably similar to https://github.com/flutter/flutter/issues/119193

You also need to check your PATH variable in .zshrc file and see if it is pointing to correct installation.

**Comment by Jayasubin** [comment]
@darshankawar I remember the path set in .zshrc file to be correct. 

Also I think that error is not relevant as the root cause was that the connection to server (for downloading dart sdk) timed out.

I tried removing flutter completely and reinstalling, but that was also not possible because the download timed out.

However, the servers came back to normal the next day, so I was able to download and setup flutter.
